pandas/io/excel/_base.py

```diff
@@ -888,32 +888,77 @@ def close(self):
         return content
 
 
-def _is_ods_stream(stream: Union[BufferedIOBase, RawIOBase]) -> bool:
+def _peek_stream(
+    stream: Union[BufferedIOBase, RawIOBase, BinaryIO], size: int = 20
+) -> bytes:
     """
-    Check if the stream is an OpenDocument Spreadsheet (.ods) file
-
-    It uses magic values inside the stream
+    Return the specified number of bytes from the start of the stream
+    and seek back to the start of the stream afterwards.
 
     Parameters
     ----------
     stream : Union[BufferedIOBase, RawIOBase]
-        IO stream with data which might be an ODS file
+    size: int
+      The number of bytes to read.
 
     Returns
     -------
-    is_ods : bool
-        Boolean indication that this is indeed an ODS file or not
+    content : bytes
+        The bytes founds.
     """
     stream.seek(0)
```

Comment:
What if stream doesn't support seek?  I think this is possible with compressed files.

Reply:
I have a feeling we'd have seen reports of this either in the pandas tracker or xlrd's tracker, since, unless I'm mistaken this type of seeking has been used in xlrd for many years now.